cpp/benchmarks/common/cudf_tpch_datagen/tpch_datagen.hpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/table/table.hpp>
+
+#include <string>
+#include <vector>
+
+namespace CUDF_EXPORT cudf {
+namespace datagen {
+namespace schema {
+
+auto const ORDERS   = std::vector<std::string>{"o_orderkey",

```

Comment:
Why are these vectors global? Do they need to be defined in the .hpp (are they used in multiple translation units)?

Reply:
Actually I can't find a single other use of `ORDERS` in this PR (?).

Reply:
I actually need to export these vectors to the user's of this library. If they want to write out the generated tables to disk, these vectors provide them with the schema of the table.

Reply:
Sorry, I'm not following why these need to be in the header. Are the vectors written disk as is or are certain fields selected during generation?
If the former, could we moved these to the .cpp and provide an accessor instead? Example:
```
std::vector<std::string> get_orders_vector();
or
std::vector<std::string> get_orders_vector( range_selection ); // initializer list, vector of indices, start/end index
```

Reply:
Agree, put these behind an accessor. They could even be function-local static, as long as you make sure to EXPORT the function.

Reply:
I actually ended up removing the schema vectors. This is also inline with the stock datagen that TPC-H provides, it leaves applying the schema to the user. I will change the code to make sure that the columns in the generated table are in the same order as in the TPC-H specs.